sztp-agent/pkg/dhcp/network_manager.go

```diff
@@ -0,0 +1,60 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+Copyright (C) 2022-2023 Intel Corporation
+Copyright (c) 2022 Dell Inc, or its subsidiaries.
+Copyright (C) 2022 Red Hat.
+*/
+
+// Package dhcp implements the DHCP client
+package dhcp
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/godbus/dbus/v5"
+)
+
+// GetBootstrapURLViaNetworkManager returns the sztp redirect URL via NetworkManager
+func GetBootstrapURLViaNetworkManager() (string, error) {
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return "", fmt.Errorf("failed to connect to system bus: %v", err)
+	}
+
+	nm := conn.Object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
+
+	var primaryConnPath dbus.ObjectPath
+	err = nm.Call("org.freedesktop.DBus.Properties.Get", 0, "org.freedesktop.NetworkManager", "PrimaryConnection").Store(&primaryConnPath)
```

Comment:
Why only primary connection? I think we should loop over all active connections…